### ISSUE TITLE
All Wallets in device are not displaying to connect wallet (Looking for Caesium wallet to connect from wallet connect)

### ISSUE THREAD

**Issue by nagarjuna-caesium** [WalletConnect/WalletConnectUnity#141]
**Bug
- Only some wallets from device are loading

**To Reproduce**
1. Click on Connect to Connect Wallet Connect from third web service
2. Click on Deep Link when QR code is available
3.  On Clicking Deep link Button only Some wallets are showing to choose

**Expected behavior**
My expection is too load all wallets in the device or atleast it should Display our CAESIUM wallet

**Environment**
 - OS: Android
 - Unity version [e.g. 2022.3.12f1]

**This happening in Unity build for Android**

**Wallets used for testing**
 [e.g. Metamask iOS, Trust Android]

**Additional context**
Please help me how to lauch Caesium wallet when connected with deep link

**Comment by nagarjuna-caesium** [comment]
HI 
Im using Thirdweb SDK and trying to connect from WALLET CONNECT to our new product CAESIUM wallet which is available in play store and also in Explore list from WalletConnect Caesium is available

Im Using this code to connect from third web sdk, and it is showing Wallet Connect prefab and Deeplink button also working and can able to choose other wallets(some wallets in device).

//////////
     thirdSDK = ThirdwebManager.Instance.SDK;

        // Configure the connection
        var connection = new WalletConnection(
          provider: WalletProvider.WalletConnect,       // The wallet provider you want to connect to (Required)
          chainId: 56                               // The chain you want to connect to (Required)
          
        );
/////////
and intilizing Caesium wallet also from third web

 walletConnectExplorerRecommendedWalletIds =
                    (walletConnectExplorerRecommendedWalletIds == null || walletConnectExplorerRecommendedWalletIds.Length == 0 || string.IsNullOrEmpty(walletConnectExplorerRecommendedWalletIds[0]))
                        ? new string[]
                        {
                            "c57ca95b47569778a828d19178114f4db188b89b763c899ba0be274e97267d96", // metamask
                            "4622a2b2d6af1c9844944291e5e7351a6aa24cd7b23099efac1b2fd875da31a0", // trustwallet
                            "225affb176778569276e484e1b92637ad061b01e13a048b35a9d280c3b58970f", // safe
                           
                            "9aa288a4c66884bec28af54056921660ebde9debd848ef03902363ed534c96d4", //caesium

**Comment by nagarjuna-caesium** [comment]
We are using Thirdweb Unity Sdk that is avilable here
https://portal.thirdweb.com/unity

-- Following instructions from the give nportal only
https://portal.thirdweb.com/unity/wallet/walletconnect

This is working with other Main Wallets like Metamask and trust but Caesium is not showing when trying to to connect from Deeplink( Dapp URL), from Caesium App we QR code is scanning and working fine.

But here major issue is From wallet Connect our Caesium App is not showing to select a wallet before connecting.Please help me out on this

**Comment by skibitsky** [comment]
WalletConnect Modal works well with Caesium; the problem must be in the Thirdweb SDK. Please open an issue in the Thirdweb repository or contact them directly.

**Comment by nagarjuna-caesium** [comment]
Wallet Connect is working with Caesium but while try to connect a wallet from app chooser Caesium is not showing
![WhatsApp Image 2023-12-12 at 11 39 05 AM](https://github.com/WalletConnect/WalletConnectUnity/assets/146815051/c8694015-41c6-4fb9-880a-811b540a1d54)

**Comment by skibitsky** [comment]
This is the repository for the official WalletConnect packages for Unity. Since you are using the Thirdweb SDK, please open an issue in the Thirdweb SDK repository or contact Thirdweb directly.

**Comment by nagarjuna-caesium** [comment]
Even the same Issue we are facing with Wallet Connect unity3d SDk v2.1.0
<img width="1792" alt="wallet connect v2 1 0" src="https://github.com/WalletConnect/WalletConnectUnity/assets/146815051/f5500596-0c28-4802-997b-9ce7d3df9e5c">

**Comment by skibitsky** [comment]
No such issue in v3.0.0, I tested it on Android with Caesium wallet. See my [original response](https://github.com/WalletConnect/WalletConnectUnity/issues/141#issuecomment-1862187702).

**Comment by nagarjuna-caesium** [comment]
Can you please provide the link for v3.0.0 SDK

**Comment by nagarjuna-caesium** [comment]
Do we need to copy all three folders from the package in to Unity Project Assets folder?
<img width="1276" alt="Screenshot 2023-12-21 at 12 58 29 PM" src="https://github.com/WalletConnect/WalletConnectUnity/assets/146815051/74dcc4a5-213c-4fda-9c68-471a7a22c5a5">

**Comment by skibitsky** [comment]
You can find installation instructions in the [docs](https://docs.walletconnect.com/advanced/walletconnectmodal/about?platform=unity).

Once packages are installed, you can import sample from the Modal package. The sample includes prefab and sample scene.

**Comment by nagarjuna-caesium** [comment]
Ok will check it with installing walletconnect modal
https://github.com/WalletConnect/WalletConnectUnity/tree/main/Packages/com.walletconnect.modal

and with wallet connect Sdk can we connect to wallet and can we load NFT items too?

**Comment by skibitsky** [comment]
Yes, but you would need to manually implement interaction with smart contracts using Nethereum. This is a new package, and we are currently focusing on stable wallet connections and session requests.

We [have docs](https://docs.walletconnect.com/advanced/walletconnectmodal/usage?platform=unity#interaction-with-rpc) on how to implement smart contract interactions with Nethereum and WalletConnect Modal.

The flow would look like this:
1. The user connects their wallet with WalletConnect Modal.
2. The dapp sends a `personal_sign` request; the user signs the message in the wallet.
3. The dapp receives the signature in response. It uses [Nethereum to validate the signature](https://nethereum.readthedocs.io/en/latest/nethereum-signing-messages/#2-verifying-a-signed-message-encoded-in-utf8-using-encodeutf8andecrecover). This step is necessary to ensure that the user actually owns the address.
4. At this point, the dapp knows that the user is the owner of the address and can make `eth_call` requests to RPC nodes to load the NFT balance from the smart contract. This is similar to how the `eth_getBalance` request is implemented [in our documentation](https://docs.walletconnect.com/advanced/walletconnectmodal/usage?platform=unity#interaction-with-rpc). Here is a [StackExchange post](https://ethereum.stackexchange.com/a/102522) that can be useful. Alternatively, you can use a blockchain explorer API. Some of them provide convenient ways to load NFTs owned by an address.

**Comment by nagarjuna-caesium** [comment]
I have included all three folders in the unity project but from package it is showing as scripts missing for all prefab items as shown in the screenshot please help me out from this.

<img width="1790" alt="Screenshot 2023-12-22 at 11 13 04 AM" src="https://github.com/WalletConnect/WalletConnectUnity/assets/146815051/29c2fc2e-637d-4aa9-9884-054e3ac18945">

**Comment by skibitsky** [comment]
Please follow installation instructions. Packages aren't supposed to be manually included into `Assets` folder.
